### PR TITLE
`storage` and `storage_cache` - update tests for 4.0

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

File: internal/services/storage/storage_table_entity_data_source_test.go [hunk 1]
```diff
@@ -72,10 +71,9 @@ func (d StorageTableEntityDataSource) basicWithDataSource(data acceptance.TestDa
 %s
 
 data "azurerm_storage_table_entity" "test" {
-  table_name           = azurerm_storage_table_entity.test.table_name
-  storage_account_name = azurerm_storage_table_entity.test.storage_account_name
-  partition_key        = azurerm_storage_table_entity.test.partition_key
-  row_key              = azurerm_storage_table_entity.test.row_key
+  storage_table_name = azurerm_storage_table_entity.test.id
```
Comment (mbfrahry) [inline internal/services/storage/storage_table_entity_data_source_test.go:74]: Should this be `storage_table_id`

**Review by mbfrahry — APPROVED** [review]
Looks good but one comment on one of the tests

File: internal/services/storage/storage_table_entity_data_source_test.go (hunk 1 above)
Comment (catriona-m) [reply internal/services/storage/storage_table_entity_data_source_test.go:74]: yep! thank you!

**Comment by stephybun** [comment]
Superseded by #26947 and #26949

**Pull request closed without merging**